### PR TITLE
[FR] Add Migration guide

### PR DESCRIPTION
**Pull request by maririos**
Following the work that @catalinaperalta did in Python :) 
It also removes `new FormRecognizerClientOptions(FormRecognizerClientOptions.ServiceVersion.V2_1)` from the V2 samples as it is no longer needed.

I will add the link to the Migration Guide to Readme, Samples, etc in a future PR

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+- Models introduced in the latest version of the library, such as `AnalyzeResult`, remove hierarchical dependencies between document elements and move them to a more top level and easily accessible position.
+- The Form Recognizer service has further improved how to define where elements are located on documents by moving towards `BoundingRegion` definitions allowing for cross-page elements.
+- Document element fields are returned with more information, such as content and spans. 
+
+New features provided by the `DocumentModelAdministrationClient` include:
+- Users can now assign their own model IDs and specify a description when building, composing, or copying models.
+- Listing models now includes both prebuilt and custom models.
+- When using `GetModel()`, users can get the field schema (field names and types that the model can extract) for the model they specified, including for prebuilt models. 
+- Ability to get information from model operations that occurred in the last 24h.
```
Comment (kristapratico) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md:33]: super nit: last 24 hours.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md [hunk 2]
```diff
@@ -0,0 +1,739 @@
...
+
+AnalyzeDocumentOperation operation = await client.StartAnalyzeDocumentFromUriAsync("prebuilt-invoice", fileUri, options);
+
+await operation.WaitForCompletionAsync();
+
+AnalyzeResult result = operation.Value;
+
+// To see the list of all the supported fields returned by service and its corresponding types for the
+// prebuilt-invoice model, consult:
+// https://aka.ms/formrecognizer/invoicefields
```
Comment (kristapratico) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md:227]: invoicefieldschema

**Review by kristapratico — APPROVED** [review]
yayayayay 🚀

File: sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md (hunk 2 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md]: This was updated in master, so I had to update my branch.
Thanks for the reminder!

**Review by catalinaperalta — APPROVED** [review]
LGTM!

File: sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md [hunk 3]
```diff
@@ -0,0 +1,737 @@
...
+- The Form Recognizer service has further improved how to define where elements are located on documents by moving towards `BoundingRegion` definitions allowing for cross-page elements.
+- Document element fields are returned with more information, such as content and spans. 
+
+New features provided by the `DocumentModelAdministrationClient` include:
+- Users can now assign their own model IDs and specify a description when building, composing, or copying models.
+- Listing models now includes both prebuilt and custom models.
+- When using `GetModel()`, users can get the field schema (field names and types that the model can extract) for the model they specified, including for prebuilt models. 
+- Ability to get information from model operations that occurred in the last 24 hours.
+
+The below table describes the relationship of each client and its supported API version(s):
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md:35]: nit: the table below?

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md:35]: resolved in https://github.com/Azure/azure-sdk-for-net/pull/24398

File: sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md [hunk 4]
```diff
@@ -0,0 +1,737 @@
...
+
+This guide is intended to assist in the migration to `Azure.AI.FormRecognizer (4.0.x)` from versions `3.1.x` and below. It will focus on side-by-side comparisons for similar operations between versions. Please note that version `4.0.0-beta.1` will be used for comparison with `3.1.1`.
+
+Familiarity with `Azure.AI.FormRecognizer (3.1.x and below)` package is assumed. For those new to the Azure Form Recognizer client library for .NET please refer to the [README][readme] rather than this guide.
+
+## Table of Contents
+- [Migration benefits](#migration-benefits)
+- [Important changes](#important-changes)
+    - [Client usage](#client-usage)
+    - [Analyzing document](#analyzing-documents)
```
[the remark this answers is not in the document]
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/MigrationGuide.md:11]: resolved in https://github.com/Azure/azure-sdk-for-net/pull/24398

**Pull request merged**